File: Questão-14/Singleton.py
class Configuracao:
    __instance = None
    def __new__(cls):
        if Configuracao.__instance is None:
            Configuracao.__instance  = super().__new__(cls)
        return Configuracao.__instance
    
    def __init__(self):
        self.tema = "Escuro"
        
    
config = Configuracao()
config2 = Configuracao()

config.tema = "Claro"

print (config2.tema)

config.tema = "Escuro Avermelhado"

print (config2.tema)
        
    
        

File: Questão-14/Singleton.go
package main

import (
	"fmt"
	"sync"
)

type Configuracao struct{
	tema string
}

var instance *Configuracao
var once sync.Once

func GetConfiguracao() *Configuracao{
	once.Do(func(){
		instance = &Configuracao{
			tema: "Escuro",
		}
	})
	return instance
}

func (c *Configuracao) AlterarTema(tema string){
	c.tema = tema

}

func main(){
	config := GetConfiguracao()
	config2 := GetConfiguracao()

	config.AlterarTema("Claro")

	fmt.Println(config2.tema)

	config.AlterarTema("Vermelho Escuro")

	fmt.Println(config2.tema)

}